### PR TITLE
docs: add blog on deploy tabby to huggingface space

### PR DESCRIPTION
**Pull request by wsxiaoys**
cc @randxie

<s>TBD:

1. T4 GPU
2. install vscode extension
3. authentication
4. clone an existing space
</s>

image

### CODE REVIEW

File: website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ENV HOME=/home/user \
+    PATH=/home/user/.local/bin:$PATH
+
+ENV TABBY_ROOT=$HOME/app
+
+WORKDIR $HOME/app
+
+COPY --chown=user . $HOME/app
+
+CMD ["serve", "--model", "TabbyML/SantaCoder-1B"]
```
Comment (randxie) [inline website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md:110]: This should be:

```
"--device", "cuda", "--model", "TabbyML/CodeLlama-7B", "--compute-type", "int8"
```

I tried with the CPU version. The app can be brought up but very slow. Using GPU as default is a good idea given that it's a 7B model.

**Comment by randxie** [comment]
For the `edit-dockerfile.png`, could you update the screenshot to this one?

image

I have verified other steps. After switching to T4 GPU, the request latency is much better.

File: website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+authors:
+  - name: Rand Xie
+    title: Contributor of Tabby
+    url: https://github.com/randxie
+    image_url: https://github.com/randxie.png
+  - meng
+tags: [deployment]
+---
+
+[Hugging Face Spaces](https://huggingface.co/spaces) provide an easy way for anyone to host their machine learning models.
```
Comment (gyxlucy) [inline website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md:13]:
```suggestion
[Hugging Face Spaces](https://huggingface.co/spaces) provides an easy way for anyone to host their machine learning models.
```

File: website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+That’s it! With the hosted APIs, you will be able to call these APIs in your apps. Next, we will deep dive into each step with screenshots!!
+**Everything will be done in the Hugging Face UI. No local setup is needed.**
+
+### Deep Dive
+
+#### Create a new Space
+
+After you create a Hugging Face account, you should be able to see the following page by clicking this [link](https://huggingface.co/new-space).
+The owner name will be your account name. Fill in a Space name, e.g. "tabbyml", and select Docker as Space SDK. Then click "Create Space" at the bottom.
```
Comment (randxie) [inline website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md:37]: You might want to mention using "T4" GPU here. By default, newly created space uses 2-Core CPU.

**Comment by randxie** [comment]
LGTM!!

File: website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+![Create a new Space](./new-space.png)
```
Comment (wsxiaoys) [inline website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md:39]: TBD: replace image with `Space hardware` set to `T4` GPU

File: website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+Click on the “App” button, you should be able to see the container is building:
+
+![Space Building](./building.png)
+
+If the App is up successfully, you should see this page:
+
+![Tabby Swagger](./tabby-swagger.png)
+
+#### Call the API to do Code Completion
```
Comment (gyxlucy) [inline website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md:83]:
```suggestion
#### Call code completion API
```

File: website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+For advanced users, you can leverage the Git workspace. In this blog post, we will show you the UI flow instead. After you click the "Create a Space" in the last step, you will be directed to this page. Just ignore the main text and click the "Files" on the top right corner.
+
+![Docker Space](./docker-space.png)
+
+After clicking on the "Files", you will be able to see a "Add file" button, click that, then click on "Create a new file"
+
+![Empty Space](./empty-space.png)
+
+Then you will be redirected to the page below. Set the filename to “Dockerfile” and copy the content to the “Edit” input box. You can copy the code from the [appendix](#dockerfile) here. Once ready, click the button “Commit new file to main” on the bottom.
```
Comment (gyxlucy) [inline website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md]:
```suggestion
Then you will be redirected to the page below. Set the filename to “Dockerfile” and copy the content to the “Edit” input box. You can copy the code from the [appendix](#dockerfile) here to bring up the [SantaCoder-1B](https://huggingface.co/TabbyML/SantaCoder-1B) model. Once ready, click the button “Commit new file to main” on the bottom.
```

**Pull request merged**